src/rewrite_reduce.cpp

```diff
@@ -54,9 +54,71 @@ struct find_softmax
     }
 };
 
+struct find_reduce_mean
+{
+    auto matcher() const { return match::name("reduce_mean"); }
+
+    void apply(module& m, const match::matcher_result& r) const
+    {
+        auto reduce_mean = r.result;
+        auto op          = reduce_mean->get_operator().to_value();
+        auto axes        = op["axes"].to_vector<std::int64_t>();
+        auto input       = reduce_mean->inputs().front();
+
+        bool is_integral = false;
+        double max_n     = 0;
+        input->get_shape().visit_type([&](auto t) {
+            is_integral = t.is_integral();
+            max_n       = t.max();
+        });
+
+        auto dims = input->get_shape().lens();
+        double n  = std::accumulate(
+            axes.begin(), axes.end(), 1, [&](const auto& prod_value, const auto& axis) {
+                auto norm_axis = axis < 0 ? dims.size() + axis : axis;
+                return prod_value * dims.at(norm_axis);
+            });
+
+        // avoid overflow (the larger value will be later handled)
+        if(n >= max_n / 4)
+            return;
+
+        auto n_literal = m.add_literal(literal{{input->get_shape().type(), {1}}, {n}});
+        if(is_integral)
+        {
+            auto out_lens = reduce_mean->get_shape().lens();
+            for(const auto axis : axes)
+            {
+                auto norm_axis      = axis < 0 ? out_lens.size() + axis : axis;
+                out_lens[norm_axis] = 1;
+            }
+
+            auto reduce_sum =
+                m.insert_instruction(reduce_mean, make_op("reduce_sum", {{"axes", axes}}), input);
+            auto n_bcast = m.insert_instruction(
+                reduce_mean, make_op("multibroadcast", {{"out_lens", out_lens}}), n_literal);
+            auto div = m.insert_instruction(reduce_mean, make_op("div"), {reduce_sum, n_bcast});
```

Comment:
Use `insert_common_op(reduce_mean, make_op("div"), {reduce_sum, n_literal})` instead of inserting `multibroadcast` manually.